go2ts.go

```diff
@@ -96,38 +128,90 @@ func (g *Go2TS) tSTypeFromStructFieldType(reflectType reflect.Type) *tsType {
 		kind = reflectType.Kind()
 	}
 
-	// Default to setting the tsType from the Go type.
-	ret.typeName = reflectTypeToTypeScriptType(reflectType)
+	// As we build up the chain of tsTypes that fully describes a typeDefinition
+	// we may come across named types. For example: map[string]Donut, where
+	// Donut could be a "type Donut struct {...}", or a type based on a
+	// primitive type, such as "type Donut string". In this case we need to add
+	// that type to all of our known types and return a reference to that type
+	// from here.
+	if !calledFromAddType && // Don't do this if called from addType().
+		reflectType.Name() != "" && // Don't bother with anonymous structs.
```

Comment:
It would be useful for the reader to understand why we don't bother with anonymous structs and time.Time. A brief rationale would be great.

Reply:
Ah, the anonymous struct comment was wrong, fixed.
Also updated comment to point to explanation of time.Time issue.